BLL/PaginatedList.cs

```diff
@@ -44,10 +46,10 @@ public string IsOnCurrentPage(int pageIndex, int currentPageIndex)
             }
         }
 
-        public static PaginatedList<T> CreatePaginatedList(IQueryable<T> source, int pageIndex, int pageSize)
+        public static async Task<PaginatedList<T>> CreatePaginatedList(IQueryable<T> source, int pageIndex, int pageSize)
         {
-            var count = source.Count();
-            var items = source.Skip((pageIndex - 1) * pageSize).Take(pageSize).ToList();
+            var count = await source.CountAsync();
+            var items = await source.Skip((pageIndex - 1) * pageSize).Take(pageSize).ToListAsync();
             return new PaginatedList<T>(items, count, pageIndex, pageSize);
         }
```

Comment:
Make sure that those changes are required, they don't seem to be part of this PR subject.